lib/src/connection.rs

```diff
@@ -53,9 +64,47 @@ impl Connection {
         user: &str,
         password: &str,
     ) -> Result<Connection> {
+        let root_cert_store = Self::build_cert_store();
+        let stream = Self::build_stream(stream, host, root_cert_store).await?;
+
+        Self::init(user, password, stream).await
+    }
+
+    async fn new_tls_with_certificate<T: AsRef<str>>(
+        stream: TcpStream,
+        host: &Host<T>,
+        user: &str,
+        password: &str,
+        certificate: &ClientCertificate,
+    ) -> Result<Connection> {
+        let mut root_cert_store = Self::build_cert_store();
+
+        let cert_file = File::open(certificate.cert_file.as_str())?;
+        let mut reader = BufReader::new(cert_file);
+        let certs = rustls_pemfile::certs(&mut reader);
+        for certificate in certs {
+            if let Ok(cert) = certificate {
+                root_cert_store.add(cert).unwrap();
+            }
+        }
```

Comment:
I'd rather avoid the unwrap and ignore the certificate:

```suggestion
        let certs = rustls_pemfile::certs(&mut reader).flatten();
        root_cert_store.add_parsable_certificates(certs);
```